db/migrate/20190913125905_migrate_subjects.rb

```diff
@@ -0,0 +1,13 @@
+class MigrateSubjects < ActiveRecord::Migration[5.2]
+  def change
+    say_with_time 'populating/migrating course subjects' do
+      all_subject = Subject.all
+      all_course_includes_ucas_subjects = Course.includes(:ucas_subjects, :provider, :subjects)
+      all_course_includes_ucas_subjects.each do |course|
+        course.level = course.ucas_level
+        course.subjects = all_subject.where subject_name: course.dfe_subjects.map(&:to_s)
```

Comment:
Changing `where` to a `select` will ensure that the db isn't hit. I can't guarantee it was being, with `where`, but looking at the log file it does look like it probably was.

```suggestion
        course.subjects = all_subject.select { |s| s.subject_name.in? course.dfe_subjects.map(&:to_s) }
```